### ISSUE TITLE
Support for web worker, H265, single frame extraction

### ISSUE THREAD

**Issue by siebeneicher** [gpac/mp4box.js#76]
Hi,

first of all, great work on mp4box!

Without really digging into the documentation, I have a couple of questions and I hope you can give me a short reply.

Do you plan in near future to support H265?
Can mp4box.js be used in a web worker (so no deps to the DOM, like video or canvas element)?
Can mp4box.js extract the binary data of a single frame, like using canvas.getImageData()? Could we choose which frames get extracted individually?
Is mp4box.js stream ready, so we don't need to wait to load the whole movie before starting to extract the first frames?

Would really appreciate a short reply from you!

Take care,

Markus

**Comment by siebeneicher** [comment]
I digged into the code and successfully got the samples of a video. Nice and easy.

So I could answer some of my questions:

* yes, it streams
* yes, you get the raw data of the samples

So I still wonder how hard it would be to decode the samples, that I can get for each sample a fully decoded image usable for drawing on a canvas. I know I could append the data to use media source extension and let a video element do the decoding and grab each frame an image via canvas.getImageData(). But that requires a video element, therefore only available right now in the main thread.

How hard would it be to extend the library with a JS decoder for lets say H264?

**Comment by cconcolato** [comment]
Thanks for your interest. To answer your questions:
- Yes, HEVC is supported. You can check [this demo page](http://download.tsi.telecom-paristech.fr/gpac/mp4box.js/hevcimageviewer/)
- Regarding Web Worker, mp4box does not have dependency on the DOM. Actually, it's used in a Service Worker. Check [this demo](https://github.com/gpac/mp4box-sw).
- Extraction of a given sample is possible, but the data will be compressed and not decoded, so not really like Canvas.
- Regarding streaming, yes MP4Box works in streaming mode. You just pass the data as it arrives and you'll get the extracted samples or the segments as they come.

**Comment by siebeneicher** [comment]
Hey thanks for your reply. I think I will dig more deep into and see whats possible. My target is to use HVEC + web worker + specifi c sample range decoding. The last one (specifi c sample range decoding) is needed because I might need to jump to a different time position, so I would need to decode from that position on forward.

**Comment by cconcolato** [comment]
mp4box.js has a seek function in the API. Open a new issue if it does not work.
